kyuubi-server/src/main/scala/org/apache/kyuubi/server/statestore/jdbc/JDBCStateStore.scala

```diff
@@ -0,0 +1,438 @@
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import com.google.common.annotations.VisibleForTesting
+import com.zaxxer.hikari.{HikariConfig, HikariDataSource}
+
+import org.apache.kyuubi.{KyuubiException, Logging, Utils}
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.config.KyuubiConf._
+import org.apache.kyuubi.server.statestore.StateStore
+import org.apache.kyuubi.server.statestore.api.Metadata
+import org.apache.kyuubi.server.statestore.jdbc.DatabaseType._
+
+class JDBCStateStore(conf: KyuubiConf) extends StateStore with Logging {
+  import JDBCStateStore._
+
+  private val dbType = DatabaseType.withName(conf.get(SERVER_STATE_STORE_JDBC_DATABASE_TYPE))
+  private val driverClassOpt = conf.get(SERVER_STATE_STORE_JDBC_DRIVER)
+  private val driverClass = dbType match {
+    case DERBY => driverClassOpt.getOrElse("org.apache.derby.jdbc.AutoloadedDriver")
+    case MYSQL => driverClassOpt.getOrElse("com.mysql.jdbc.Driver")
+    case CUSTOM => driverClassOpt.getOrElse(
+        throw new IllegalArgumentException("No jdbc driver defined"))
+  }
+
+  private val databaseAdaptor = dbType match {
+    case DERBY => new DerbyDatabaseDialect
+    case MYSQL => new MysqlDatabaseDialect
+    case CUSTOM => new GenericDatabaseDialect
+  }
+
+  private val datasourceProperties = new Properties()
+  conf.getStateStoreJDBCDataSourceProperties.foreach { case (key, value) =>
+    datasourceProperties.put(key, value)
+  }
+
+  private val hikariConfig = new HikariConfig(datasourceProperties)
+  hikariConfig.setDriverClassName(driverClass)
+  hikariConfig.setJdbcUrl(conf.get(SERVER_STATE_STORE_JDBC_URL))
+  hikariConfig.setUsername(conf.get(SERVER_STATE_STORE_JDBC_USER))
+  hikariConfig.setPassword(conf.get(SERVER_STATE_STORE_JDBC_PASSWORD))
+  hikariConfig.setPoolName("kyuubi-state-store-pool")
+
+  @VisibleForTesting
+  private[kyuubi] val hikariDataSource = new HikariDataSource(hikariConfig)
+  private val mapper = new ObjectMapper().registerModule(DefaultScalaModule)
+
+  if (conf.get(SERVER_STATE_STORE_JDBC_DATABASE_SCHEMA_INIT)) {
+    initSchema()
+  }
+
+  private def initSchema(): Unit = {
+    val classLoader = getClass.getClassLoader
+    val initSchemaStream: Option[InputStream] = dbType match {
+      case DERBY =>
+        Option(classLoader.getResourceAsStream("sql/derby/statestore-schema-derby.sql"))
+      case MYSQL =>
+        Option(classLoader.getResourceAsStream("sql/mysql/statestore-schema-mysql.sql"))
+      case CUSTOM => None
+    }
+    initSchemaStream.foreach { inputStream =>
+      try {
+        val statements = new BufferedReader(new InputStreamReader(inputStream)).lines()
+          .collect(Collectors.joining("\n")).trim.split(";")
+        withConnection() { connection =>
+          Utils.tryLogNonFatalError {
+            statements.foreach { statement =>
+              execute(connection, statement)
+              info(s"Execute init schema query: $statement successfully.")
```

Comment:
```
try {
   ...
} finally {
  statement.close
}
```

Reply:
here the statement is the sql query string.
let me rename it.